### PR TITLE
Introduce flag to disable usage of clients and tenants

### PR DESCRIPTION
**Pull request by rinckm**
### Description

Steward is providing a client/tenant concept which can be used.
It is also possible to use Steward without clients and tenants and use plain k8s namespaces instead. 
This change provides a possibility to disable the usage of tenants and clients in the helm chart.
This results in less resource consumption on the k8s cluster by saving one controller.

### Submitter checklist

- [x] Change has been tested (on a back-end cluster)
- [x] (If applicable) Jira backlog item ID added to the PR title and the [changelog.yaml] entry
- [x] [changelog.yaml] entry with upgrade notes is prepared and appropriate for the audience affected by the change (users or developer, depending on the change)
- [x] Semantic version diffed against [last release][releases] and updated accordingly. In this project the version has to be maintained here:
    - [/charts/steward/Chart.yaml](https://github.com/SAP/stewardci-core/blob/master/charts/steward/Chart.yaml) (`version` and `appVersion`)

In case dependencies have been updated:
- [ ] Links to external changelogs, since the last release of our component, added to the [changelog.yaml] entry (description).
- [ ] Changelogs read thoroughly, potential impact described, upgrade notes prepared (if necessary)
- [ ] Check if dependency updates affect our semantic version increment.

### Reviewer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There is at least 1 approval for the pull request and no outstanding requests for change
- [ ] All voter checks have passed
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] The Pull Request title is understandable and reflects the changes well
- [ ] The Pull Request description is understandable and well documented
- [ ] [changelog.yaml] entry for this Pull Request has been added
    - [ ] Changelog entry contains all required information
    - [ ] 'Upgrade notes' are documented in changelog.yaml (if required)

[changelog.yaml]: https://github.com/SAP/stewardci-core/changelog.yaml
[releases]: https://github.com/SAP/stewardci-core/releases

### CODE REVIEW

File: charts/steward/templates/clusterrole-client.yaml [hunk 1]
```diff
@@ -8,3 +9,4 @@ rules:
 - apiGroups: ["steward.sap.com"]
   resources: ["tenants"]
   verbs: ["create","delete","get","list","patch","update","watch"]
+{{- end }}
```
Comment (rinckm) [inline charts/steward/templates/clusterrole-client.yaml:12]:
```suggestion
{{- end }}

```

File: charts/steward/templates/clusterrole-tenant-controller.yaml [hunk 2]
```diff
@@ -23,3 +24,4 @@ rules:
   resources: ["podsecuritypolicies"]
   verbs:     ["use"]
   resourceNames: [{{ include "steward.tenantController.podSecurityPolicyName" . | quote }}]
+{{- end }}
```
Comment (rinckm) [inline charts/steward/templates/clusterrole-tenant-controller.yaml:27]:
```suggestion
{{- end }}

```

File: charts/steward/templates/clusterrole-tenant.yaml [hunk 3]
```diff
@@ -11,3 +12,4 @@ rules:
 - apiGroups: [""]
   resources: ["secrets"]
   verbs: ["create","delete","get","list","patch","update","watch"]
+{{- end }}
```
Comment (rinckm) [inline charts/steward/templates/clusterrole-tenant.yaml:15]:
```suggestion
{{- end }}

```

File: charts/steward/templates/clusterrolebinding-tenant-controller.yaml [hunk 4]
```diff
@@ -10,3 +11,4 @@ subjects:
 - kind: ServiceAccount
   name: steward-tenant-controller
   namespace: {{ .Values.targetNamespace.name | quote }}
+{{- end }}
```
Comment (rinckm) [inline charts/steward/templates/clusterrolebinding-tenant-controller.yaml:14]:
```suggestion
{{- end }}

```

File: charts/steward/templates/deployment-tenant-controller.yaml [hunk 5]
```diff
@@ -79,3 +80,4 @@ spec:
       tolerations:
         {{- toYaml . | nindent 8 }}
       {{- end }}
+{{- end }}
```
Comment (rinckm) [inline charts/steward/templates/deployment-tenant-controller.yaml:83]:
```suggestion
{{- end }}

```

File: charts/steward/README.md [hunk 6]
```diff
@@ -124,6 +124,7 @@ The tables in the following sections list the configurable parameters of the Ste
 
 | Parameter | Description | Default |
 |---|---|---|
+| <code>tenantController.<wbr/><b>enabled</b></code><br/><i>bool</i> | Switch for enabling tenant controller and use of client and tenants. | false |
```
Comment (holgpar) [inline charts/steward/README.md:127]: The default seems to be "true".

Comment (holgpar) [inline charts/steward/README.md:127]:
```suggestion
| <code>tenantController.<wbr/><b>enabled</b></code><br/><i>bool</i> | Switch for enabling tenant controller and use of clients and tenants. | false |
```

**Review by holgpar — APPROVED** [review]

**Review by anbrsap — APPROVED** [review]

**Pull request merged**